src/qt/guiutil.cpp

```diff
@@ -1064,7 +1065,36 @@ void loadStyleSheet(QWidget* widget, bool fDebugWidget)
                 if (!qFile.open(QFile::ReadOnly)) {
                     throw std::runtime_error(strprintf("%s: Failed to open file: %s", __func__, file.toStdString()));
                 }
-                stylesheet->append(QLatin1String(qFile.readAll()));
+
+                QString strStyle = QLatin1String(qFile.readAll());
+                // Process all <os=...></os> groups in the stylesheet first
+                QRegularExpressionMatch osStyleMatch;
+                QRegularExpression osStyleExp("^(<os=(?:'|\").+(?:'|\")>)((?:.|\n)+?)(</os>?)$");
+                osStyleExp.setPatternOptions(QRegularExpression::MultilineOption);
+                QRegularExpressionMatchIterator it = osStyleExp.globalMatch(strStyle);
+
+                // For all <os=...></os> sections
+                while (it.hasNext() && (osStyleMatch = it.next()).isValid()) {
+                    QStringList listMatches = osStyleMatch.capturedTexts();
+
+                    // Full match + 3 group matches
+                    if (listMatches.size() % 4) {
+                        throw std::runtime_error(strprintf("%s: Invalid <os=...></os> section in file %s", __func__, file.toStdString()));
+                    }
+
+                    for (int i = 0; i < listMatches.size(); i += 4) {
+                        if (!listMatches[i + 1].contains(QString::fromStdString(BitcoinGUI::DEFAULT_UIPLATFORM))) {
```

Comment:
Shouldn't this respect `-uiplatform`? i.e. smth like a3c6d534a8

Reply:
ha, good point 👍 picked!